tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/management/ActiveMQServerControlTest.java

```diff
@@ -2769,27 +2775,554 @@ public void testListAllConsumersAsJSON() throws Exception {
+      Assert.assertTrue(second.getString(ConsumerField.SESSION.getAlternativeName()).length() > 0);
+      Assert.assertEquals(((ClientSessionImpl) session2).getName(), second.getString(ConsumerField.SESSION.getAlternativeName()));
+      Assert.assertTrue(second.getString(ConsumerField.QUEUE.getAlternativeName()).length() > 0);
+      Assert.assertEquals(queueName.toString(), second.getString(ConsumerField.QUEUE.getAlternativeName()));
+      Assert.assertEquals(false, second.getBoolean(ConsumerField.BROWSE_ONLY.getName()));
+      Assert.assertEquals(0, second.getJsonNumber(ConsumerField.MESSAGES_IN_TRANSIT.getName()).longValue());
+      Assert.assertEquals(0, second.getJsonNumber(ConsumerField.MESSAGES_IN_TRANSIT_SIZE.getName()).longValue());
+      Assert.assertEquals(0, second.getJsonNumber(ConsumerField.MESSAGES_ACKNOWLEDGED.getName()).longValue());
+      Assert.assertEquals(0, second.getJsonNumber(ConsumerField.LAST_DELIVERED_TIME.getName()).longValue());
+      Assert.assertEquals(0, second.getJsonNumber(ConsumerField.LAST_ACKNOWLEDGED_TIME.getName()).longValue());
 
-      Assert.assertTrue(second.getJsonNumber("creationTime").longValue() > 0);
-      Assert.assertNotNull(second.getJsonNumber("consumerID").longValue());
-      Assert.assertTrue(second.getString("connectionID").length() > 0);
-      Assert.assertEquals(factory2.getConnection().getID().toString(), second.getString("connectionID"));
-      Assert.assertTrue(second.getString("sessionID").length() > 0);
-      Assert.assertEquals(((ClientSessionImpl) session2).getName(), second.getString("sessionID"));
-      Assert.assertTrue(second.getString("queueName").length() > 0);
-      Assert.assertEquals(queueName.toString(), second.getString("queueName"));
-      Assert.assertEquals(false, second.getBoolean("browseOnly"));
-      Assert.assertEquals(0, second.getJsonNumber("deliveringCount").longValue());
+   }
+
+
+   @Test
+   public void testListAllConsumersAsJSONTXCommit() throws Exception {
+      SimpleString queueName = new SimpleString(UUID.randomUUID().toString());
+      ActiveMQServerControl serverControl = createManagementControl();
+
+      ServerLocator locator = createInVMNonHALocator();
+      ClientSessionFactory factory = createSessionFactory(locator);
+      ClientSession session = factory.createSession(true,false, 1);
+      addClientSession(session);
+
+      serverControl.createAddress(queueName.toString(), RoutingType.ANYCAST.name());
+      if (legacyCreateQueue) {
+         server.createQueue(queueName, RoutingType.ANYCAST, queueName, null, false, false);
+      } else {
+         server.createQueue(new QueueConfiguration(queueName).setRoutingType(RoutingType.ANYCAST).setDurable(false));
+      }
+
+      ClientConsumer consumer = session.createConsumer(queueName, null, 100, -1, false);
+      addClientConsumer(consumer);
+      session.start();
+      Thread.sleep(200);
```

Comment:
Necessary? Repeat comment for other tests below

Reply:
fixed